### PR TITLE
enh: calcmedian interface

### PR DESCRIPTION
**Pull request by mgxd**
Changes proposed in this pull request
- ENH: `CalculateMedian` interface that takes one or more 4D realigned timeseries and returns the average across each timeseries' medians.

### CODE REVIEW

File: nipype/algorithms/misc.py [hunk 1]
```diff
@@ -1380,6 +1380,51 @@ def merge_rois(in_files, in_idxs, in_ref,
     return out_file
 
 
+class CalculateMedianInputSpec(BaseInterfaceInputSpec):
+    in_file = InputMultiPath(File(exists=True, mandatory=True,
```
Comment (satra) [inline nipype/algorithms/misc.py]: `in_files` since multiple files are acceptable.

File: nipype/algorithms/misc.py [hunk 2]
```diff
@@ -1380,6 +1380,51 @@ def merge_rois(in_files, in_idxs, in_ref,
...
+      desc="One or more realigned Nifti 4D timeseries"))
+    median_file = traits.Str('median.nii.gz', usedefault=True,
+      desc="Filename to store median image")
```
Comment (satra) [inline nipype/algorithms/misc.py]: perhaps add a boolean flag to indicate whether the median should be computed individually or across files. and generate the appropriate outputs based on the flag.

File: nipype/algorithms/misc.py [hunk 3]
```diff
@@ -1380,6 +1380,51 @@ def merge_rois(in_files, in_idxs, in_ref,
     return out_file
 
 
+class CalculateMedianInputSpec(BaseInterfaceInputSpec):
+    in_file = InputMultiPath(File(exists=True, mandatory=True,
+      desc="One or more realigned Nifti 4D timeseries"))
+    median_file = traits.Str('median.nii.gz', usedefault=True,
```
Comment (satra) [inline nipype/algorithms/misc.py]: use name_source and template

Comment (mgxd) [reply nipype/algorithms/misc.py]: I defined a `_gen_fname` instead that should handle multiple `in_files` accordingly

**Pull request merged**